src/parser/statement.js

```diff
@@ -899,7 +918,7 @@ export default class StatementParser extends ExpressionParser {
       }
     }
 
-    if (decorators.length) {
+    if (!this.state.inCommaSeparatedFields && decorators.length) {
```

Comment:
I don't understand why `inCommaSeparatedFields` affects this. Maybe it should be a separate check instead (if fields can't have a trailing comma)?